src/main/kotlin/org/rust/ide/inspections/lints/RsNamingInspection.kt

```diff
@@ -109,8 +127,8 @@ open class RsUpperCaseNamingInspection(elementType: String) : RsNamingInspection
 
     override fun checkName(name: String): Pair<Boolean, String?> {
         val str = name.trim('_')
-        if (!str.isEmpty() && str.all { !it.isLetter() || it.isUpperCase() }) {
-            return Pair(true, null)
+        if (str.isNotEmpty() && str.all { !it.hasCase || !it.isLetter() || it.isUpperCase() }) {
```

Comment:
```suggestion
        if (str.isNotEmpty() && str.none { it.isLowerCase() }) {
```

I think it does the same but more readable